### ISSUE TITLE
Enhancement: make torch optional — DR + metrics in core, torch/lightning in a [lightning] extra

### ISSUE THREAD

**Issue by cmvcordova** [latent-reasoning-works/manylatents#287]
## Summary

`torch>=2.3` and `lightning>=2.5` sit in manylatents' **core** `dependencies`, so anyone who
wants `pca` / `umap` / `tsne` / `leiden` or the metric registry installs the entire training
stack. Measured, most of that is avoidable: **12 of 17 latent algorithms import and run with
torch hard-blocked**, and the metric registry needs it for almost nothing.

Proposal: keep **one distribution**, move torch/lightning into a `[lightning]` extra, leave DR
and metrics in core.

This is an enhancement, not a bug report — nothing is broken today. Filing it because a
downstream consumer (`geomancer`) cannot require manylatents without inflicting a 1.3 GB
install, and half its recipe catalogue needs manylatents for the **DR itself**, not the metrics.

## Measured

All figures from executing against `main`, not from reading:

| | now | with the split |
|---|---|---|
| install | 94 packages, **1.3 GB** | 50 packages, **426 MB** |
| `import manylatents.algorithms.latent` | **1.134s** | **0.016s** |

**Torch-free with torch hard-blocked (12 of 17):** `pca`, `umap`, `tsne`, `leiden`, `phate`,
`diffusion_map`, `mds`, `multiscale_phate`, `reeb_graph`, `classifier`, `trajectory_aligner`,
`dr_noop`. Metrics: **63 of 87 reachable names compute torch-free** — the 24 failures are
missing-dataset/module errors, not torch.

**Irreducible (1):** `aa` — `archetypes 0.6.2` declares `torch` in its own metadata, and
nothing here can undo that. Suggests an `[archetypes]` extra.

**Genuinely neural (4):** `merging`, `foundation_encoder`, `gpu_phate_local`,
`selective_correction`.

**Migration size:** 55 files carry a module-level `import torch`. **39 are already on the
`[lightning]` side**; the core-side work is the 16 in `algorithms/latent` plus 4 shared
support files.

## Why most of it is cheap

`LatentModule` is a plain `ABC`, not a `LightningModule`. Most algorithms touch torch for
exactly one purpose — matching the OUTPUT type to the INPUT type, so a caller who passed a
Tensor gets one back:

```python
def _to_numpy(x):       return x.detach().cpu().numpy() if isinstance(x, Tensor) else np.asarray(x)
def _to_output(r, ref): return torch.tensor(r, ...) if isinstance(ref, Tensor) else ...
```

That is a courtesy to torch callers, not a requirement of PCA.

## Three traps, found by building it wrong first

I prototyped this and **reverted it**. Recording the failures so nobody repeats them:

1. **`latent_module_base.py` is not the wall.** `algorithms/latent/__init__.py` imports
   `phate.py`, which imports torch at line 5. Making the base module lazy changes *zero*
   public import paths. The lazy `__init__` is a prerequisite for any version of this.

2. **Stripping `from torch import Tensor` while leaving `Tensor` annotations breaks the module
   silently.** `dr_noop.py` raised `NameError` at import, `_build_registry`'s bare
   `except Exception` swallowed it, `list_algorithms()` went **15 → 14**, and **34 tests still
   passed** — the suite asserts three names and never a count.

3. **`ArrayLike = Union[np.ndarray, Tensor]` is a public alias.** Rewriting it to reference
   `Any` breaks `get_type_hints()` — the ForwardRef resolves in the *consumer's* namespace, and
   `trajectory_aligner.py` doesn't import `Any`.

Also: a `sys.modules.get("torch")` probe for "is this a Tensor" is *almost* sound, but returns
False when torch is evicted from `sys.modules` while a live Tensor exists — reachable via
`mock.patch.dict` and via the torch-blocking CI job this work needs. Prefer
`hasattr(x, "detach") and hasattr(x, "cpu")`.

## Costs, stated squarely

- **It is subtractive.** Consumers silently lose torch on their next sync. That is a breaking
  change wearing a dependency tweak's clothes and wants a major/minor decision, not a quiet
  merge.
- **No repo tests torch-free today.** CI installs `.[all]` and 28 of 81 test files import torch
  at module scope. Without a torch-blocked job the property rots the day after it lands.

## The spec

Full write-up, 541 lines with every claim tagged to `file:line` or the command that produced
it, including a "Not verified" section:

`geomancer/docs/scoping/manylatents-torch-optional.md`

It also corrects an earlier geomancer-side spec (`manylatents-metric-driver.md` §4.2), which
recommended a separate distribution on the premise that *"every algorithm subclasses
`LatentModule`, which imports torch at module scope."* That premise is measured false, and §4.2
is the one paragraph in that document carrying no `[RAN]` tag.

Happy to open a PR if the direction is wanted — but the subtractive-change call and the CI job
are decisions for whoever owns this repo, not mine to assume.
